pkg/certificate/reconciler.go

```diff
@@ -0,0 +1,35 @@
+package certificate
+
+import (
+	"context"
+	"time"
+)
+
+// CheckAndUpdate checks for a MRC to reach a desired condition and updates the status when the condition is met
+func (r *MRCReconciler) CheckAndUpdate(ctx context.Context, mrcClient MRCClient, checkInterval time.Duration) {
+	ticker := time.NewTicker(checkInterval)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				ticker.Stop()
+				log.Info().Msgf("Stopping status reconciliation loop for MRC %s", r.mrcName)
+				return
+			case <-ticker.C:
+				mrc := mrcClient.Get(r.mrcName)
+				if mrc != nil {
```

Comment:
```suggestion
				if mrc == nil {
```

Also, we probably want to `continue` after this step....we need to decide how to handle errors that occur at this level. With a kubernetes error like this (while retrieving an MRC), I'm inclined to just keep retrying, but I'm interested in others' opinions

Reply:
Oops, yep, this should be `==` rather than `!=`